sinks/graphite.py

```diff
@@ -32,77 +43,84 @@ def __init__(self, host="localhost", port=2003, prefix="statsite.", attempts=3,
         if protocol not in ["pickle", "lines"]:
             raise ValueError("Supported protocols are pickle, lines")
 
+        if normalize is not None and normalize not in ("False", "false", "No", "no"):
+            self.normalize_func = self.normalize_key
+        else:
+            self.normalize_func = lambda k: "%s%s" % (self.prefix, k)
+
         self.logger = logging.getLogger("statsite.graphitestore")
         self.host = host
         self.port = port
         self.prefix = prefix
         self.attempts = attempts
         self.sock = self._create_socket()
         self.flush = self.flush_pickle if protocol == "pickle" else self.flush_lines
+        self.metrics = []
 
-    def flush_lines(self, metrics):
+    def normalize_key(self, key):
         """
-        Flushes the metrics provided to Graphite.
+        Take a single key string and return the same string with spaces, slashes and
+        non-alphanumeric characters subbed out and prefixed by self.prefix.
+        """
+        key = SPACES.sub("_", key)
+        key = SLASHES.sub("-", key)
+        key = NON_ALNUM.sub("", key)
+        key = "%s%s" % (self.prefix, key)
+        return key
 
-       :Parameters:
-        - `metrics` : A list of "key|value|timestamp" strings.
+    def append(self, metric):
         """
-        if not metrics:
-            return
+        Add one metric to queue for sending. Addtionally modify key to be compatible with txstatsd
+        format.
 
-        self.logger.info("Outputting %d metrics", len(metrics))
+        :Parameters:
+         - `metric` : A single statsd metric string in the format "key|value|timestamp".
+        """
+        if metric and metric.count("|") == 2:
+            k, v, ts = metric.split("|")
+            k = self.normalize_func(k)
+            self.metrics.append(((k), v, ts))
```

Comment:
ah, yeah not needed, just an artifact from the earlier revision which had different logic here.